app/src/main/java/com/juliazluo/www/pokedex/ListActivity.java

```diff
@@ -0,0 +1,78 @@
+package com.juliazluo.www.pokedex;
+
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.util.TypedValue;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CompoundButton;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.Switch;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+public class ListActivity extends AppCompatActivity {
+
+    protected static PokemonHandler pokemonHandler;
+    protected ArrayList<Pokedex.Pokemon> pokemonList;
+    protected FilteredListAdapter adapter;
+    protected Switch layoutSwitch;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list);
+        final RecyclerView recyclerView = (RecyclerView) findViewById(R.id.filter_recycler);
+        pokemonHandler = new PokemonHandler();
+        pokemonList = new ArrayList<>();
+        ArrayList<Pokedex.Pokemon> copy = pokemonHandler.getPokemonList();
+        pokemonList.addAll(copy);
+        adapter = new FilteredListAdapter(getApplicationContext(), pokemonList);
+        recyclerView.setAdapter(adapter);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+
+        layoutSwitch = (Switch) findViewById(R.id.layout_switch);
+        layoutSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                if (isChecked) {
+                    recyclerView.setLayoutManager(new GridLayoutManager(getApplicationContext(), 4));
+
+
+                } else {
+                    recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+                }
+
+            }
+        });
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Intent intent = getIntent();

```

Comment:
By executing this in onResume, it will rerandomize even if I go to a pokemon's profile activity and come back to the listactivity. That means I can only really view one pokemon in the random activity before it is randomized again.